### PR TITLE
Remove `haskell.bzl`

### PR DESCRIPTION
**Pull request by guibou**
`haskell.bzl` is deprecated, but still wrong.

### CODE REVIEW

File: haskell/haskell.bzl [hunk 1]
```diff
@@ -13,9 +13,12 @@ load(
     _haskell_repl_aspect = "haskell_repl_aspect",
     _haskell_test = "haskell_test",
     _haskell_toolchain = "haskell_toolchain",
-    _haskell_toolchain_libraries = "haskell_toolchain_libraries",
     _haskell_toolchain_library = "haskell_toolchain_library",
 )
+load(
+    ":private/haskell_impl.bzl",
+    _haskell_toolchain_libraries = "haskell_toolchain_libraries",
```
Comment (aherrmann) [inline haskell/haskell.bzl]: It's an internal rule that shouldn't be user facing. Why not remove it from this file altogether?
See https://github.com/tweag/rules_haskell/pull/1045/files

Comment (guibou) [reply haskell/haskell.bzl]: I totally agree to remove it if you want.

Comment (aherrmann) [reply haskell/haskell.bzl]: Yes, I think that's better.

Comment (guibou) [reply haskell/haskell.bzl]: Done.

Comment (aherrmann) [reply haskell/haskell.bzl]: > Why not remove it from this file altogether?

I meant to remove `haskell_toolchain_libraries` from `haskell/haskell.bzl`. On the other hand, [release 0.10](https://github.com/tweag/rules_haskell/pull/1073/files) will contain the deprecation of `haskell/haskell.bzl`, so it seems fine to me to have that file removed completely in the release that follows after. cc @Profpatsch

Comment (Profpatsch) [reply haskell/haskell.bzl]: Should this PR here go into 0.10 then? Or 0.11?

Comment (aherrmann) [reply haskell/haskell.bzl]: This PR should then go into 0.11.

**Pull request merged**